### PR TITLE
[Reachability] Scan summary and better docs

### PR DESCRIPTION
**Pull request by meghfossa**
# Overview

This PR, 
- Adds scan summary for reachability analysis
- Adds better docs for reachability, and moves reachability docs in better location

## Acceptance criteria

- Scan summary is displayed for reachability

## Testing plan

```bash
git checkout feat/better-docs && make install-dev

git clone https://github.com/fossas/reachability-with-maven-example.git
cd reachability-with-maven-example

# (required) perform maven build
# if jar artifact is not present reachability will not be performed
mvn package

# Analyze with fossa-dev
fossa-dev analyze -p reachability-example-better-docs -r example-revision --fossa-api-key <MY-FOSSA-KEY>
```

- You should see the scan summary in stdout.
- You should see `OrgId` does not support reachability message, if your org does not support reachability

## Risks

N/A

## Metrics

N/A

## References

https://fossa.atlassian.net/browse/ANE-1416

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this PR added docs, I added links as appropriate to the user manual's ToC in `docs/README.ms` and gave consideration to how discoverable or not my documentation is.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json` AND I have updated example files used by `fossa init` command. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).
- [x] If I made changes to a subcommand's options, I updated `docs/references/subcommands/<subcommand>.md`.

### CODE REVIEW

File: src/App/Fossa/Reachability/Types.hs [hunk 1]
```diff
@@ -70,3 +72,9 @@ data ParsedJar = ParsedJar
   deriving (Eq, Ord, Show, Generic)
 
 instance ToJSON ParsedJar
+
+reachabilityRawJson :: Text
```
Comment (meghfossa) [inline src/App/Fossa/Reachability/Types.hs:76]: To avoid circular imports

File: docs/README.md [hunk 2]
```diff
@@ -59,6 +59,7 @@ Concept guides explain the nuances behind how basic FOSSA primitives work. If yo
   - [Analyzing Specific Submodules](./walkthroughs/analysis-target-configuration.md#target-filtering-for-submodules)
 - [Dynamic Strategy Command Selection](./features/strategy-command-selection.md)
 - [`fossa analyze` Reference](./references/subcommands/analyze.md)
+- [Reachability](./features/reachability.md)
```
Comment (jssblck) [inline docs/README.md:62]: [optional] consider explicitly denoting this as "vulnerability reachability" or similar.

File: src/App/Fossa/Analyze/ScanSummary.hs [hunk 3]
```diff
@@ -341,6 +364,15 @@ skippedDueFilter = ": skipped (exclusion filters)"
...
 skippedDueNonProductionPathFiltering = ": skipped (non-production path filtering)"
 
+skippedReachabilityDueToPartialGraph :: Doc AnsiStyle
+skippedReachabilityDueToPartialGraph = ": skipped (partial graph)"
+
+skippedReachabilityDueToNotSupported :: Doc AnsiStyle
+skippedReachabilityDueToNotSupported = ": skipped (not supported)"
+
+skippedReachabilityDueToMissingAnalysis :: Doc AnsiStyle
+skippedReachabilityDueToMissingAnalysis = ": skipped (no dependency analysis)"
```
Comment (jssblck) [inline src/App/Fossa/Analyze/ScanSummary.hs:374]: Consider linking to the doc in comments with a note saying to update the doc if these change.

File: src/App/Fossa/Analyze/Upload.hs [hunk 4]
```diff
@@ -111,10 +111,7 @@ uploadSuccessfulAnalysis (BaseDir basedir) metadata jsonOutput revision scanUnit
 
     if (orgSupportsReachability org)
       then void $ upload revision metadata reachabilityUnits
-      else
-        unless (null reachabilityUnits) $
-          logDebug . pretty $
-            "Organization: (" <> show (organizationId org) <> ") does not support reachability! skipping reachability analysis upload!"
+      else logDebug . pretty $ "Organization: (" <> show (organizationId org) <> ") does not support reachability! skipping reachability analysis upload!"
```
Comment (jssblck) [inline src/App/Fossa/Analyze/Upload.hs:114]: [optional] I love more personality in my tools but I've firsthand witnessed corporate users demeaning tools that aren't, well, dry and corporate.

If the goal is emphasis using color or font weighting doesn't, IME, trigger these kinds of issues.
```suggestion
      else logDebug . pretty $ "Organization: (" <> show (organizationId org) <> ") does not support reachability. Skipping reachability analysis upload."
```

File: src/App/Fossa/Reachability/Upload.hs [hunk 5]
```diff
@@ -51,12 +53,11 @@ analyzeForReachability ::
...
-  AnalysisScanResult ->
-  m [SourceUnitReachability]
-analyzeForReachability analysisResult = context "reachability" $ do
-  let analyzerResult = analyzersScanResult analysisResult
-  units <- catMaybes <$> (traverse callGraphOf analyzerResult)
-  debugMetadata reachabilityRawJson units
+  [DiscoveredProjectScan] ->
+  m [SourceUnitReachabilityAttempt]
+analyzeForReachability analyzerResult = context "reachability" $ do
+  units <- (traverse callGraphOf analyzerResult)
```
Comment (jssblck) [inline src/App/Fossa/Reachability/Upload.hs:59]: [nit] are these parens necessary? surprised linter didn't cry
```suggestion
  units <- traverse callGraphOf analyzerResult
```

**Review by jssblck — APPROVED** [review]

**Pull request merged**